dotnet/src/SemanticKernel.Core/Functions/KernelFunctionFromMethod.cs

```diff
@@ -399,6 +408,55 @@ private static bool IsAsyncMethod(MethodInfo method)
         return (parameterFunc, parameterView);
     }
 
+    /// <summary>
+    /// Tries to deserialize the given value into an object of the specified target type.
+    /// </summary>
+    /// <param name="value">The value to be deserialized.</param>
+    /// <param name="targetType">The type of the object to deserialize the value into.</param>
+    /// <param name="deserializedValue">The deserialized object if the method succeeds; otherwise, null.</param>
+    /// <returns>true if the value is successfully deserialized; otherwise, false.</returns>
+    private static bool TryToDeserializeValue(object value, Type targetType, out object? deserializedValue)
+    {
+        try
+        {
+            // Handling known 'JSON' types.
+            if (value is JsonDocument document)
+            {
+                deserializedValue = JsonSerializer.Deserialize(document, targetType);
+                return true;
+            }
+
+            if (value is JsonNode node)
+            {
+                deserializedValue = JsonSerializer.Deserialize(node, targetType);
+                return true;
+            }
+
+            if (value is JsonElement element)
+            {
+                deserializedValue = JsonSerializer.Deserialize(element, targetType);
+                return true;
+            }
+
+            // The JSON can be represented by other data types from various libraries. For example, JObject, JToken, and JValue from the Newtonsoft.Json library.  
+            // Since we don't take dependencies on these libraries and don't have access to the types here,
+            // the only way to deserialize those types is to convert them to a string first by calling the 'ToString' method.
+            deserializedValue = JsonSerializer.Deserialize(value.ToString(), targetType);
```

Comment:
Did you consider JsonSerializer.Serialize(value) instead of value.ToString()? That may or may not be better. Or maybe it could be yet another fallback.

Reply:
Anyway, the Duck Typing capability has been reverted for now, until we have more information to make an educated decision.